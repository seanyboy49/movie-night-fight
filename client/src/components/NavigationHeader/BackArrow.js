import React from 'react'
import { Link } from 'react-router-dom'

import leftArrowImg from '../../images/arrow-left.svg'

const BackArrow = ({ backLink }) => {
  // placeholder div to evenly space sibling elements in flex parent
  if (!backLink) {
    return <div style={{ width: '2rem', height: '2rem' }}></div>
  }
  return (
    <Link to={backLink}>
      <img src={leftArrowImg} alt="" />
    </Link>
  )
}

export default BackArrow
