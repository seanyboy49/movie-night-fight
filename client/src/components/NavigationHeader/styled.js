import styled from 'styled-components'

export const AppTitle = styled.h1`
  font-family: bevan;
  font-size: 25px;
  color: ${(props) => props.color || 'black'};
  margin: 0;
`

export const NavContainer = styled.div`
  background-color: ${(props) => props.background || 'white'};
  display: flex;
  padding: 1rem;
  align-items: center;
  justify-content: center;
  border-bottom: 2px solid ${(props) => props.background || '#e5e5e5'};

  > * {
    margin-left: auto;
  }
`

export const Nav = styled.div`
  width: 100%;
  height: 10vh;
`
