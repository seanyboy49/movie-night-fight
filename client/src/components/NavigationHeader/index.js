import React, { useState } from 'react'
import DropdownMenu from '../DropdownMenu'

import Hamburger from '../Hamburger'
import { NavContainer, AppTitle } from './styled'

const NavigationHeader = () => {
  const [isOpen, setIsOpen] = useState(false)

  return (
    <>
      <NavContainer>
        <AppTitle>Movie Night Fight</AppTitle>

        <Hamburger onClick={setIsOpen} isOpen={isOpen} />
      </NavContainer>
      <DropdownMenu isOpen={isOpen} />
    </>
  )
}

export default NavigationHeader
