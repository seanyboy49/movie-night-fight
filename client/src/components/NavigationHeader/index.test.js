import React from 'react'
import { shallow } from 'enzyme'

import NavigationHeader from '.'

describe('<NavigationHeader />', () => {
  it('should render', () => {
    const wrapper = shallow(<NavigationHeader />)

    expect(wrapper).toIncludeText('Movie Night Fight')
    expect(wrapper).toMatchSnapshot()
  })
})
