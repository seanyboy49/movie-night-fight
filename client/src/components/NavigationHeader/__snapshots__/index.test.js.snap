// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationHeader /> should render 1`] = `
<Fragment>
  <styled.div>
    <styled.h1>
      Movie Night Fight
    </styled.h1>
    <Burger
      isOpen={false}
      onClick={[Function]}
    />
    <DropdownMenu
      isOpen={false}
    />
  </styled.div>
</Fragment>
`;
